erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -995,85 +995,84 @@ def get_unconsumed_raw_materials(self):
-				and se.work_order= %s and ifnull(sed.t_warehouse, '') != ''
-			group by sed.item_code, sed.t_warehouse
-		""", self.work_order, as_dict=1)
-
-		materials_already_backflushed = frappe.db.sql("""
-			select
-				item_code, sed.s_warehouse as warehouse, sum(qty) as qty
-			from
-				`tabStock Entry` se, `tabStock Entry Detail` sed
-			where
-				se.name = sed.parent and se.docstatus=1
-				and (se.purpose='Manufacture' or se.purpose='Material Consumption for Manufacture')
-				and se.work_order= %s and ifnull(sed.s_warehouse, '') != ''
-			group by sed.item_code, sed.s_warehouse
-		""", self.work_order, as_dict=1)
-
-		backflushed_materials= {}
-		for d in materials_already_backflushed:
-			backflushed_materials.setdefault(d.item_code,[]).append({d.warehouse: d.qty})
-
-		po_qty = frappe.db.sql("""select qty, produced_qty, material_transferred_for_manufacturing from
-			`tabWork Order` where name=%s""", self.work_order, as_dict=1)[0]
-
-		manufacturing_qty = flt(po_qty.qty)
-		produced_qty = flt(po_qty.produced_qty)
-		trans_qty = flt(po_qty.material_transferred_for_manufacturing)
+		qty_to_manufacture, produced_qty = frappe.db.get_value("Work Order", self.work_order, ["qty", "produced_qty"])
+		qty_to_manufacture = flt(qty_to_manufacture)
+		produced_qty = flt(produced_qty)
+		remaining_qty = qty_to_manufacture - produced_qty
+
+		query = """
+			SELECT
+				sed.item_name,
+				sed.original_item,
+				sed.item_code,
+				sum(sed.qty) AS qty,
+				sed.t_warehouse AS warehouse,
+				sed.description,
+				sed.stock_uom,
+				sed.expense_account,
+				sed.cost_center
+			FROM
+				`tabStock Entry` se,
+				`tabStock Entry Detail` sed
+			WHERE
+				se.name = sed.parent
+					AND se.docstatus = 1
+					AND se.purpose in %s
+					AND se.work_order = %s
+					AND ifnull(sed.t_warehouse, '') != ''
+			GROUP BY
+				sed.item_code,
+				sed.t_warehouse
+		"""
+
+		# get all items already consumed for manufacture against the work order
+		consumed_materials = frappe.db.sql(query, (['Manufacture', 'Material Consumption for Manufacture'], self.work_order), as_dict=1)
+		backflushed_materials = {}
+		for item in consumed_materials:
+			backflushed_materials.setdefault(item.item_code, []).append({item.warehouse: item.qty})
 
+		# loop through the transferred quantities, and add an entry for each
+		# remaining item needed for manufacture, if not already consumed
+		transferred_materials = frappe.db.sql(query, (['Material Transfer for Manufacture'], self.work_order), as_dict=1)
 		for item in transferred_materials:
-			qty= item.qty
 			item_code = item.original_item or item.item_code
-			req_items = frappe.get_all('Work Order Item',
+			transferred_qty = item.qty
+			transferred_qty_each = flt(transferred_qty / qty_to_manufacture)
+
+			required_item = frappe.get_all("Work Order Item",
 				filters={'parent': self.work_order, 'item_code': item_code},
 				fields=["required_qty", "consumed_qty"]
 				)
 
```

Comment:
What if user has transferred extra materials which is not there in the BOM Item?

Reply:
I've made a fix so that it defaults the consumed quantity to zero, if it doesn't exist in the BOM or WO.